apps/css-workshop/src/components/Checkbox.astro

```diff
@@ -69,3 +67,9 @@ const { status, isLoading, indeterminate, disabled, variant, basic, ...props } =
     </label>
   )
 }
+
+<script>
+  const indeterminateInputs = [...document.querySelectorAll('input[data-iui-indeterminate]')];
+
+  indeterminateInputs.forEach((input) => (input.indeterminate = true));
```

Comment:
This script was added as a way to resolve [a comment](https://github.com/iTwin/iTwinUI/pull/1970#discussion_r1554254285) from the original PR.

Reply:
why are you using `data-iui-indeterminate` over `indeterminate`?

Reply:
I think the issue is that the types are checking for the `data-` prefix for custom attributes, so maybe you can simplify to just `data-indeterminate`.

Reply:
You can ignore this type error using `@ts-ignore`. It's such an unnecessary thing to flag.

The `data-iui-` prefix is reserved for attributes that are meaningfully used in our CSS.

Reply:
switched it over and added the TS ignore

Reply:
Why do you have multiple `@ts-ignore`s? You only need one next to the `indeterminate` attribute.

Reply:
It looks like you can actually just use `as astroHTML.JSX.InputHTMLAttributes` instead of `@ts-ignore`. And when you do `querySelectorAll`, you need to give it a generic `HTMLInputElement` type. See full code below

<details><summary>Checkbox.astro</summary>

```astro
---
type Props = {
  wrapperProps?: astroHTML.JSX.HTMLAttributes;
  isLoading?: boolean;
  indeterminate?: boolean;
  variant?: 'eyeball';
  status?: 'negative' | 'positive' | 'warning';
} & astroHTML.JSX.InputHTMLAttributes;

const {
  class: className,
  disabled,
  status,
  isLoading,
  indeterminate,
  variant,
  wrapperProps,
  ...props
} = Astro.props;

const basic = !Astro.slots.has('default');

const inputProps = {
  ['class:list']: [
    'iui-checkbox',
    {
      'iui-loading': isLoading,
      'iui-checkbox-visibility': variant === 'eyeball',
    },
    className,
  ],
  indeterminate: indeterminate,
  type: 'checkbox',
  disabled: disabled,
  ...props,
} as astroHTML.JSX.InputHTMLAttributes;
---

{basic && <input {...inputProps} />}
{
  !basic && (
    <label
      class:list={[
        'iui-checkbox-wrapper',
        {
          'iui-disabled': disabled,
          'iui-loading': isLoading,
          [`iui-${status}`]: !!status,
        },
      ]}
      {...wrapperProps}
    >
      <input {...inputProps} />
      {isLoading && (
        <div
          class='iui-progress-indicator-radial'
          data-iui-indeterminate='true'
          data-iui-size='x-small'
        />
      )}
      <span class='iui-checkbox-label'>
        <slot />
      </span>
    </label>
  )
}

<script>
  const indeterminateInputs = [
    ...document.querySelectorAll<HTMLInputElement>('input[type=checkbox][indeterminate]'),
  ];
  indeterminateInputs.forEach((input) => {
    input.indeterminate = true;
  });
</script>
```
</details>

Reply:
missed this comment. push up that change in the last push